vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -70,10 +73,92 @@ public class Popover extends Component implements HasAriaLabel, HasComponents {
     public Popover() {
         getElement().getNode().addAttachListener(this::attachComponentRenderer);
 
+        // Workaround for: https://github.com/vaadin/flow/issues/3496
+        getElement().setProperty("opened", false);
+
+        getElement().addPropertyChangeListener("opened", event -> {
+            // Only handle client-side changes, server-side changes are already
+            // handled by setOpened
+            if (event.isUserOriginated()) {
+                doSetOpened(this.isOpened(), event.isUserOriginated());
+            }
+        });
+
         updateTrigger();
         setOverlayRole("dialog");
     }
 
+
+    /**
+     * {@code opened-changed} event is sent when the overlay opened state
+     * changes.
+     */
+    public static class OpenedChangeEvent extends ComponentEvent<Popover> {
+        private final boolean opened;
+
+        public OpenedChangeEvent(Popover source, boolean fromClient) {
+            super(source, fromClient);
+            this.opened = source.isOpened();
+        }
+
+        public boolean isOpened() {
+            return opened;
+        }
+    }
+
+    /**
+     * Opens or closes the popover.
+     *
+     * @param opened
+     *            {@code true} to open the popover, {@code false} to close it
+     */
+    public void setOpened(boolean opened) {
+        if (opened != isOpened()) {
+            doSetOpened(opened, false);
+        }
+    }
+
+    /**
+     * Opens the popover.
+     */
+    public void open() {
+        setOpened(true);
+    }
+
+    /**
+     * Closes the popover.
+     */
+    public void close() {
+        setOpened(false);
+    }
+
+    private void doSetOpened(boolean opened, boolean fromClient) {
+        getElement().setProperty("opened", opened);
+        fireEvent(new OpenedChangeEvent(this, fromClient));
+    }
+
+    /**
+     * Gets the open state from the popover.
+     *
+     * @return the {@code opened} property from the popover
+     */
+    @Synchronize(property = "opened", value = "opened-changed")
```

Comment:
So this has a similar issue to what was fixed for Dialog in https://github.com/vaadin/flow-components/pull/5258.

This is reproducable as such:
- Configure popover to open / close on hover
- Add a button to the popover to open a modal dialog
- Clicking the button opens the dialog and closes the popover on the client
- However on the server the opened changed event for the popover is ignored due to server-side modality
- This results in the popover staying open on the server, while it is actually closed on the client

Not really sure what to do about this. For Dialog we fixed it so that a dialog doesn't close itself on the client-side on outside click if another modal dialog is opened as part of that click. That doesn't really make sense for the popover though, which should always close when it is not hovered. 

We could ignore the issue for now and document it in a ticket. I think in that case the property listener in the constructor and the `doSetOpened` method are not necessary, as those were introduced as part of the fix mentioned above.

Reply:
Thanks, I missed this. Reported #6433 and removed the property listener / `doSetOpened` method for now.